From 5f1690cf7a37c7a2e3700829386f841c0b07eb3b Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
Date: Sat, 13 May 2023 23:51:46 +0600
Subject: [PATCH 10/85] Revert "clk: renesas: r8a779f0: Remove RPC clock"

This reverts commit d43e969bc8a236f2864844cb34cdafdd8dfae39d.

On VC4, RPC is supported under Linux.
---
 drivers/clk/renesas/r8a779f0-cpg-mssr.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/clk/renesas/r8a779f0-cpg-mssr.c b/drivers/clk/renesas/r8a779f0-cpg-mssr.c
index 14bb80d3e15a..964b985ac926 100644
--- a/drivers/clk/renesas/r8a779f0-cpg-mssr.c
+++ b/drivers/clk/renesas/r8a779f0-cpg-mssr.c
@@ -48,6 +48,7 @@ enum clk_ids {
 	CLK_PLL6_DIV2,
 	CLK_S0,
 	CLK_SDSRC,
+	CLK_RPCSRC,
 	CLK_OCO,
 
 	/* Module Clocks */
@@ -77,6 +78,10 @@ static const struct cpg_core_clk r8a779f0_core_clks[] __initconst = {
 	DEF_FIXED(".sdsrc",	CLK_SDSRC,	CLK_PLL5_DIV2,	2, 1),
 	DEF_RATE(".oco",	CLK_OCO,	32768),
 
+	DEF_BASE(".rpcsrc",	CLK_RPCSRC,		CLK_TYPE_GEN4_RPCSRC, CLK_PLL5),
+	DEF_BASE(".rpc",	R8A779F0_CLK_RPC,	CLK_TYPE_GEN4_RPC, CLK_RPCSRC),
+	DEF_BASE("rpcd2",	R8A779F0_CLK_RPCD2,	CLK_TYPE_GEN4_RPCD2, R8A779F0_CLK_RPC),
+
 	/* Core Clock Outputs */
 	DEF_GEN4_Z("z0",	R8A779F0_CLK_Z0,	CLK_TYPE_GEN4_Z,	CLK_PLL2,	2, 0),
 	DEF_GEN4_Z("z1",	R8A779F0_CLK_Z1,	CLK_TYPE_GEN4_Z,	CLK_PLL2,	2, 0),
@@ -139,6 +144,7 @@ static const struct mssr_mod_clk r8a779f0_mod_clks[] __initconst = {
 	DEF_MOD("msi3",		621,	R8A779F0_CLK_MSO),
 	DEF_MOD("pcie0",	624,	R8A779F0_CLK_S0D2),
 	DEF_MOD("pcie1",	625,	R8A779F0_CLK_S0D2),
+	DEF_MOD("rpc",		629,	R8A779F0_CLK_RPCD2),
 	DEF_MOD("rtdm0",	630,	R8A779F0_CLK_S0D2),
 	DEF_MOD("rtdm1",	631,	R8A779F0_CLK_S0D2),
 	DEF_MOD("rtdm2",	700,	R8A779F0_CLK_S0D2),
-- 
2.39.2

